specification/hdinsight/resource-manager/Microsoft.HDInsight/preview/2015-03-01-preview/locations.json

```diff
@@ -338,6 +377,140 @@
+        "filterMode": {
+          "description": "The filtering mode. Effectively this can enabling or disabling the VM sizes in a particular set.",
+          "type": "string"
+        },
+        "regions": {
+          "description": "The list of regions under the effect of the filter.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "clusterFlavors": {
+          "description": "The list of cluster flavors under the effect of the filter.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "nodeTypes": {
+          "description": "The list of node types affected by the filter.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "clusterVersions": {
+          "description": "The list of cluster versions affected in Major.Minor format.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "osType": {
+          "description": "The OSType affected, Windows or Linux.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        },
+        "vmSizes": {
+          "description": "The list of virtual machine sizes to include or exclude.",
+          "type": "array",
+          "items": {
+            "type": "string"
+          }
+        }
+      }
+    },
+    "BillingResources": {
+      "description": "The billing resources.",
+      "properties": {
+        "region": {
+          "description": "The region or location.",
+          "type": "string"
+        },
+        "billingMeters": {
+          "description": "The billing meter information.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/BillingMeters"
+          }
+        },
+        "diskBillingMeters": {
+          "description": "The managed disk billing information.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/DiskBillingMeters"
+          }
+        }
+      }
+    },
+    "BillingMeters": {
+      "description": "The billing meters.",
+      "properties": {
+        "meterParameter": {
+          "description": "The virtual machine sizes.",
+          "type": "string"
+        },
+        "meter": {
+          "description": "The HDInsight meter guid.",
+          "type": "string"
+        },
+        "unit": {
+          "description": "The unit of meter, VMHours or CoreHours.",
+          "type": "string"
+        }
+      }
+    },
+    "DiskBillingMeters": {
+      "description": "The disk billing meters.",
+      "properties": {
+        "diskRpMeter": {
+          "description": "The managed disk meter guid.",
+          "type": "string"
+        },
+        "sku": {
+          "description": "The managed disk billing sku, P30 or S30.",
+          "type": "string"
+        },
+        "tier": {
```

Comment:
From the description there is a limited set of values this will accept. Would it be better to have an x-ms-enum?